### ISSUE TITLE
Missing 'region_id' in 'customer_address_entity' table after upgrading version to 2.4.1

### ISSUE THREAD

**Issue by bartoll** [magento/magento2#31642]
### Preconditions (*)

1.Magento installed in version 2.3.4 with some customers in database.
2.Decision to upgrade platform do version 2.4.1

### Steps to reproduce (*)

1.Do standard upgrade process.
2.Sign-in as a customer and try to place any order using existed data information from your address book

### Expected result (*)

1. Success
2.

### Actual result (*)

1. There is an error concerning empty field "regionId" in your address
```
[2021-01-14 00:09:14] main.CRITICAL: Placing an order with quote_id 668 is failed: Proszę sprawdzić adres dostawy. Pole "regionId" jest wymaga
ne. Uzupełnij i spróbuj ponownie. [] []
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [+] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by bartoll** [comment]
Here is my SQL workaround:
```
UPDATE
  customer_address_entity
SET
  region = LOWER(region),
  region_id = (
	SELECT
	  region_id
	FROM
	  directory_country_region
	WHERE
	  lower(customer_address_entity.region) = directory_country_region.default_name
	LIMIT 1
  )
```

**Comment by engcom-Delta** [comment]
Hi @bartoll thank you for your report, I'm not able to reproduce issue by steps you described after upgrade on clean 2.4.1
**Manual testing scenario:**
- Install magento 2.3.4
- Create customer:
![image](https://user-images.githubusercontent.com/51681379/104727004-49571f00-573d-11eb-932e-4955b956f09d.png)
- Add some product to wishlist, compare list and shopping cart as registered customer
- Upgrade to 2.4.1 using [docs](https://devdocs.magento.com/guides/v2.4/comp-mgr/cli/cli-upgrade.html)
- Sign-in as a customer
- Place order with simple products
 
**Result:**
:heavy_check_mark: Order successfully placed

Also was checked this scenario after upgrade from 2.3.6 to 2.4.1

Are my steps correct or something was missed?

**Comment by engcom-Delta** [comment]
@bartoll , we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by nickpiro** [comment]
This shouldn't be closed. This is also an issue in 2.3.6 that has NOT been resolved apparently in 2.4.1?

**Comment by nickpiro** [comment]
@engcom-Delta it doesn't happen for ALL customers. I've noticed this issue tends to happen for customers or are checking out as guest, but it definitely does happen for existing customers as well. In my scenario, It seems when a customer adds an address in the checkout process. Particularly we're using stock checkout from Luma.
